packages/utils/src/internals/psTree.ts

```diff
@@ -0,0 +1,168 @@
+import type { ChildProcess } from 'child_process';
+import { spawn } from 'child_process';
+import * as readline from 'readline';
+
+export interface ProcessInfo {
+    PPID: string;
+    PID: string;
+    STAT: string | null;
+    RSS: number;
+    COMMAND: string;
+}
+
+// Type of only the string keys in ProcessInfo.
+type ProcessInfoStringKeys = {
+    [K in keyof ProcessInfo]: ProcessInfo[K] extends string | null ? K : never;
+}[keyof ProcessInfo];
+
+/**
+ * Returns a promise that resolves with an array of ProcessInfo objects representing
+ * the children of the given PID.
```

Comment:
Please try to focus on what cannot be deduced easily from the signature (i.e., no need to mention that it returns a promise of ProcessInfo array)